### PR TITLE
Add clamp worklet

### PR DESCRIPTION
**Pull request by kowczarz**
## Summary

It's a followup PR for a discussion from https://github.com/software-mansion/react-native-reanimated/pull/4933#discussion_r1297531333

## Test plan

Examples of gyroscope and accelerometer should work the same as previous.

### CODE REVIEW

**Comment by kowczarz** [comment]
Cc. @tomekzaw @tjzel

**Review by tomekzaw — APPROVED** [review]
Thanks for this PR! ❤️

File: src/reanimated2/utils.ts [hunk 1]
```diff
@@ -32,3 +32,14 @@ export function isSharedValue<T>(value: any): value is SharedValue<T> {
   'worklet';
   return value?._isReanimatedSharedValue === true;
 }
+
+/**
+ * @param val preferred value
```
Comment (tjzel) [inline src/reanimated2/utils.ts:37]:
```suggestion
 * @param value clamped value
```

File: src/reanimated2/utils.ts
Comment (tjzel) [inline src/reanimated2/utils.ts:1]: Let's move it to `interpolate.ts` since it's an edge-case of interpolation.

**Review by tjzel — CHANGES_REQUESTED** [review]

File: src/reanimated2/utils.ts [hunk 2]
```diff
@@ -32,3 +32,14 @@ export function isSharedValue<T>(value: any): value is SharedValue<T> {
...
+ * @param min minimum allowed value
+ * @param max maximum allowed value
+ * Clamps given value within a range of values between a defined minimum bound and a maximum bound.
+ */
+export function clamp(val: number, min: number, max: number) {
```
Comment (tjzel) [inline src/reanimated2/utils.ts:42]: No need for extra abbreviation - although I do love the idea of 3-letters variables.

```suggestion
export function clamp(value: number, min: number, max: number) {
```

File: src/reanimated2/utils.ts (hunk 1 above)
Comment (tomekzaw) [reply src/reanimated2/utils.ts:37]: `clamped value` would be the result value, no? Did you mean `value to be clamped`?

Comment (tjzel) [reply src/reanimated2/utils.ts:37]: It's the value you are clamping - clamped value, same as shareable value or animated value

Comment (tjzel) [reply src/reanimated2/utils.ts:37]: But I guess value to he clamped sounds more precise :like:

**Review by piaskowyk — APPROVED** [review]
I posted just two small comments, but overall it is 👌

**Review by tjzel — APPROVED** [review]

**Pull request merged**